python/pyarrow/_compute.pyx

```diff
@@ -141,6 +153,38 @@ cdef wrap_hash_aggregate_kernel(const CHashAggregateKernel* c_kernel):
     return kernel
 
 
+cdef class RecordBatchIterator(_Weakrefable):
```

Comment:
Is there any way we can use a RecordBatchReader instead of introducing a new type which seems pretty similar?

Reply:
I'll look into this.

Reply:
I included this as a separate commit, so you could review whether this is what you had in mind.